### PR TITLE
editor: use current namespace if doc has no namespace (#317)

### LINKED ISSUE #317: IllegalStateException when creating a new kubernetes yml file without namespace
**Steps:**
1. EXEC: File > New > task.yml
2. EXEC: paste the following:
```
apiVersion: "tekton.dev/v1beta1"
kind: "Task"
metadata:
  name: "foo"
spec:
  params:
    - name: "pathToDockerFile"
      type: "string"
      description: "The path to the dockerfile to build"
      default: "/workspace/workspace/Dockerfile"
  resources:
    inputs:
      - name: "workspace"
        type: "git"
    outputs:
      - name: "buildImage"
        type: "image"
  steps:
    - name: "build-sources"
      image: "fedora"
      command:
        - /bin/bash
      args: ['-c', 'echo hello world']
```
3. ASSERT: editor gets the intellij-kubernetes toolbar

**Result:**
Red error icon in the bottom right of intellij IDEA lights up. When clicking it, the upcoming dialog would unveil the following exception happened:
```
java.lang.IllegalStateException: updated must not be null
	at com.redhat.devtools.intellij.kubernetes.model.resource.kubernetes.custom.NamespacedCustomResourceOperator.get(NamespacedCustomResourceOperator.kt:80)
	at com.redhat.devtools.intellij.kubernetes.model.ClusterResource.request(ClusterResource.kt:100)
	at com.redhat.devtools.intellij.kubernetes.model.ClusterResource.get(ClusterResource.kt:87)
	at com.redhat.devtools.intellij.kubernetes.model.ClusterResource.isOutdated(ClusterResource.kt:189)
	at com.redhat.devtools.intellij.kubernetes.editor.ResourceEditor.showNotifications(ResourceEditor.kt:145)
	at com.redhat.devtools.intellij.kubernetes.editor.ResourceEditor.access$showNotifications(ResourceEditor.kt:48)
	at com.redhat.devtools.intellij.kubernetes.editor.ResourceEditor$update$1.invoke(ResourceEditor.kt:136)
	at com.redhat.devtools.intellij.kubernetes.editor.ResourceEditor$update$1.invoke(ResourceEditor.kt:48)
	at com.redhat.devtools.intellij.kubernetes.editor.ResourceEditor$sam$java_lang_Runnable$0.run(ResourceEditor.kt)
	at com.intellij.openapi.application.impl.ApplicationImpl$1.run(ApplicationImpl.java:238)
	at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:515)
	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
	at java.base/java.lang.Thread.run(Thread.java:834)
```

### PR DESCRIPTION
**Pull request by adietish**
fixes #317

### CODE REVIEW

**Review by jeffmaury — APPROVED** [review]
LGTM

File: src/main/kotlin/com/redhat/devtools/intellij/kubernetes/model/resource/NamespacedResourceOperator.kt [hunk 1]
```diff
@@ -68,16 +66,18 @@ abstract class NamespacedResourceOperator<R : HasMetadata, C: Client>(
         }
         @Suppress("UNCHECKED_CAST")
         val typedWatcher = watcher as? Watcher<R> ?: return null
+        val inNamespace = this.namespace ?: return null
```
Comment (lstocchi) [inline src/main/kotlin/com/redhat/devtools/intellij/kubernetes/model/resource/NamespacedResourceOperator.kt:69]: At the beginning of this method there is already a check for `namespace == null`

File: src/main/kotlin/com/redhat/devtools/intellij/kubernetes/model/resource/kubernetes/custom/NamespacedCustomResourceOperator.kt [hunk 2]
```diff
@@ -38,7 +38,8 @@ class NamespacedCustomResourceOperator(
     }
 
 	override fun watch(resource: HasMetadata, watcher: Watcher<in GenericCustomResource>): Watch? {
-		return watch(resource.metadata.namespace, resource.metadata.name, watcher)
+		val inNamespace = resourceOrCurrentNamespace(resource) ?: return null
```
Comment (lstocchi) [inline src/main/kotlin/com/redhat/devtools/intellij/kubernetes/model/resource/kubernetes/custom/NamespacedCustomResourceOperator.kt:41]: return null can be omitted

**Review by lstocchi — APPROVED** [review]
LGTM

**Pull request merged**